qp_shotgun/humann2/humann2.py

```diff
@@ -84,50 +85,72 @@ def generate_humann2_analysis_commands(forward_seqs, reverse_seqs, map_file,
             % ', '.join(sn_by_rp.keys()))
 
     cmds = []
-    params = ' '.join(["%s %s" % (k, v) for k, v in viewitems(parameters)])
+    params = ['--%s "%s"' % (k, v) for k, v in viewitems(parameters) if v]
     for ffn, fn, s in samples:
         cmds.append('humann2 --input "%s" --output "%s" --output-basename '
                     '"%s" --output-format biom %s' % (ffn, join(out_dir, fn),
-                                                      s, params))
+                                                      s, ' '.join(params)))
 
     return cmds
 
 
-def get_sample_names_by_run_prefix(mapping_file):
-    """Generates a dictionary of run_prefix and sample names
+def humann2(qclient, job_id, parameters, out_dir):
+    """Run humann2 with the given parameters
 
     Parameters
     ----------
-    mapping_file : str
-        The mapping file
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to run HUMAnN2
+    out_dir : str
+        The path to the job's output directory
 
     Returns
     -------
-    dict
-        Dict mapping run_prefix to sample id
-
-    Raises
-    ------
-    ValueError
-        If there is more than 1 sample per run_prefix
+    boolean, list, str
+        The results of the job
     """
-    qiime_map = pd.read_csv(mapping_file, delimiter='\t', dtype=str,
-                            encoding='utf-8', keep_default_na=False,
-                            na_values=[])
-    qiime_map.set_index('#SampleID', inplace=True)
-
-    samples = {}
-    errors = []
-    for prefix, df in qiime_map.groupby('run_prefix'):
-        len_df = len(df)
-        if len_df != 1:
-            errors.append('%s has %d samples (%s)' % (prefix, len_df,
-                                                      ', '.join(df.index)))
-        else:
-            samples[prefix] = df.index[0]
-
-    if errors:
-        raise ValueError("You have run_prefix values with multiple "
-                         "samples: %s" % ' -- '.join(errors))
-
-    return samples
+    # Step 1 get the rest of the information need to run humann2
+    qclient.update_job_step(job_id, "Step 1 of 5: Collecting information")
+    artifact_id = parameters['input']
+    # removing input from parameters so it's not part of the final command
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fps = artifact_info['files']
+
+    # Get the artifact metadata
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    qiime_map = prep_info['qiime-map']
+
+    # Step 2 generating command humann2
+    qclient.update_job_step(job_id, "Step 2 of 5: Generating HUMANn2 command")
+    rs = fps['raw_reverse_seqs'] if 'raw_reverse_seqs' in fps else []
```

Comment:
Is this always running reverse seqs if they're present, rather than allowing the option of running? 

Maybe a feature request could be to make this optional behavior?